protocol/storage/miner.go

```diff
@@ -336,7 +335,12 @@ func (sm *Miner) acceptProposal(ctx context.Context, p *storagedeal.SignedPropos
 		},
 	}
 
-	if err = resp.Sign(sm.porcelainAPI, sm.workerAddr); err != nil {
+	workerAddr, err := sm.porcelainAPI.MinerGetWorkerAddress(ctx, sm.minerAddr)
```

Comment:
This code needs to be extracted into a method that can lookup the worker address and use it to sign a response. It could be called 3 times.

Reply:
Extracted...